.gitlab-ci.yml

```diff
@@ -52,73 +53,131 @@ variables:
+      done
 
-cargo-fmt:
+build-std:
   stage:                           check-workspace
   <<:                              *docker-env
   script:
-  - cargo fmt --verbose --all -- --check
+    - all_crates=("core" "alloc" "utils" "lang2" "lang2/macro" "cli")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo build --verbose --all-features --release;
+      done
 
-cargo-clippy-af:
+build-wasm:
   stage:                           check-workspace
   <<:                              *docker-env
   script:
-  - cargo clippy --verbose --all --all-features -- -D warnings
+    - wasm_crates=("core" "alloc" "utils" "lang2" "lang2/macro")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo build --verbose --no-default-features --release --target wasm32-unknown-unknown --manifest-path ${crate}/Cargo.toml;
+      done
 
-cargo-clippy-nf:
+test-std:
   stage:                           check-workspace
   <<:                              *docker-env
   script:
-  - cargo clippy --verbose --all --no-default-features -- -D warnings
+    - all_crates=("core" "alloc" "utils" "lang2" "lang2/macro" "cli")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo test --verbose --all-features --release;
+      done
 
-cargo-test-af:
+clippy-std:
   stage:                           check-workspace
   <<:                              *docker-env
   script:
-  - cargo test --verbose --all --all-features
+    - all_crates=("core" "alloc" "utils" "lang2" "lang2/macro" "cli")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo clippy --verbose --all-features -- -D warnings;
+      done
+  allow_failure:                   true
 
-cargo-build-nf:
+clippy-wasm:
   stage:                           check-workspace
   <<:                              *docker-env
   script:
-  - cargo build --verbose --all --no-default-features --release --target=wasm32-unknown-unknown
+    - wasm_crates=("core" "alloc" "utils" "lang2" "lang2/macro")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo clippy --verbose --manifest-path ${crate}/Cargo.toml --no-default-features -- -D warnings;
+      done
+  allow_failure:                   true
+
+fmt:
+  stage:                           check-workspace
+  <<:                              *docker-env
+  script:
+    - all_crates=("core" "alloc" "utils" "lang2" "lang2/macro" "cli")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo fmt --verbose -- --check;
+      done
+  allow_failure:                   true
+
 
 #### stage:                        examples
 
-examples-wasm:
+examples-test:
   stage:                           examples
   <<:                              *docker-env
   script:
-    - for example in $(ls examples/lang); do
-        cargo build --release --no-default-features --target=wasm32-unknown-unknown --verbose --manifest-path examples/lang/$example/Cargo.toml;
+    - for DIR in examples/lang2/*; do 
```

Comment:
I think the name `example` here was totally fine.

Reply:
ok